src/cli/peakdetector/PeakDetectorCLI.cpp

```diff
@@ -665,35 +665,31 @@ void PeakDetectorCLI::saveCSV(string setName) {
     #endif
 
 	string fileName = mavenParameters->outputdir + setName + ".csv";
-
-    CSVReports* csvreports = new CSVReports(mavenParameters->samples);
-    csvreports->setMavenParameters(mavenParameters);
-
+	if(fileName.empty()) return;
+	
     if (mavenParameters->allgroups.size() == 0 ) {
 		cout << "Writing to CSV Failed: No Groups found" << endl;
         return;
     }
 
-    if(fileName.empty()) return;
-    
-    if (mavenParameters->samples.size() == 0) return;
+    vector<mzSample*> samples = mavenParameters->samples;
+    if ( samples.size() == 0) return;
 
-    csvreports->setUserQuantType(quantitationType);
+    MavenParameters* mp = mavenParameters;
+    PeakGroup::QType t = quantitationType;
+    CSVReports::ExportType exportType = CSVReports::GroupExport;
+    int selectionFlag = -1;
+    bool includeSetNamesLine = false;
 
-    //Added to pass into csvreports file when merged with Maven776 - Kiran
-    //CLI exports the default Group Summary Matrix Format (without set Names)
-    csvreports->openGroupReport(fileName);
+    CSVReports* csvExport = new CSVReports(samples, mp, t, fileName, exportType, selectionFlag, includeSetNamesLine);
 
     for(int i=0; i<mavenParameters->allgroups.size(); i++ ) {
 		PeakGroup& group = mavenParameters->allgroups[i];
-		csvreports->addGroup(&group);
+		csvExport->addItem(&group);
```

Comment:
@IpankajI Don't you think addItem is a redundant step? In my opinion, we should only have exportGroups function which can take a vector of groups or a single group and just exports the csv?

Reply:
@sahil21 Changing!

Reply:
@sahil21 Sometimes we are exporting filtered groups and we add one group at a time based on filtering criteria, So I'm letting it be as it is.

Reply:
@IpankajI Then exportGroups function should be able to take PeakGroup or vector of PeakGroup as input, right? I'm talking about function overloading.

Reply:
@IpankajI Why do we have to make temporary vector? We can pass mavenParameters->allgroups, right?